src/shared/parsing/flux/response.ts

```diff
@@ -148,3 +149,172 @@ export const parseTables = (responseChunk: string): FluxTable[] => {
 
   return tables
 }
+
+export const parseResponseWithFromFlux = (response: string): FluxTable[] => {
+  const {
+    fluxGroupKeyUnion,
+    table,
+    table: {columnKeys: keys},
+  } = fromFlux(response)
+
+  const columnKeys = [
+    'result',
+    'table',
+    ...keys.filter(k => k !== 'result' && k !== 'table'),
+  ]
+
+  const tables = table.getColumn('table')
+  const values = table.getColumn('result')
+  let valueIndex = 0
+  let tableIndex = 0
+  const dataTypes = {
+    '': '#datatype',
+  }
+  let fluxData
+  const groupSet = new Set(['result', ...fluxGroupKeyUnion])
+  // build out the dataTypes here based on the columnKeys
+  for (let i = 0; i < columnKeys.length; i++) {
+    const column = columnKeys[i]
+    fluxData = table.getOriginalColumnType(column)
+    dataTypes[column] = fluxData
+  }
+  let columnType: any
+  let columnValues
+  // build out the columnHeaders
+  let columnHeaders = columnKeys.filter(col => {
+    columnType = table.getColumnType(col)
+    columnValues = table.getColumn(col, columnType)
+    return columnValues[valueIndex] !== undefined
+  })
```

Comment:
maybe readability could be helped with the format of
```
})


// description of what i'm doing after some whitespace
let variables = { be: decided, down: here }
{
 transforming some data in here
}

```
then operations take on some visual weighting scheme. might not work, idk